curated_tokenizers/_util.py

```diff
@@ -0,0 +1,41 @@
+from pathlib import Path
+from typing import IO
+
+from curated_tokenizers.types import FileLike
+
+
+def _check_is_io_like(f: IO):
+    if not (hasattr(f, "read") and hasattr(f, "write")):
+        raise ValueError("Object is not IO-like")
+
+
+class _wrap_file_object:
+    def __init__(self, f):
+        _check_is_io_like(f)
+        self.f = f
+
+    def __enter__(self):
+        return self.f
+
+    def __exit__(self, *args):
+        pass
```

Comment:
Don't we need to release the file resource here?

Reply:
No, because then the call site cannot use a context manager. The typical call would be something like:

```
with open(...) as f:
  spp = SentencePieceProcessor.from_file(f)
```

Reply:
I see that the `SentencePieceProcessor` docstring doesn't document this, whereas `ByteBPEProcessor` does. So I'll add a commit with this change in the docs.

Reply:
Added.